### PR TITLE
Feat/patient onboarding

### PR DESCRIPTION
**Pull request by Kushagraved**
add conditional rendering for otp

### CODE REVIEW

File: src/app/layout.jsx [hunk 1]
```diff
@@ -8,7 +10,10 @@ export default function RootLayout({ children }) {
         head.jsx. Find out more at https://beta.nextjs.org/docs/api-reference/file-conventions/head
       */}
       <head />
-      <body>{children}</body>
+      <body>    
+{children}     
+ </body>
+
```
Comment (rkdixit96) [inline src/app/layout.jsx:16]: Fix liniting

File: src/app/page.jsx [hunk 2]
```diff
@@ -1,12 +1,7 @@
-import Image from 'next/image'
-import { Inter } from '@next/font/google'
-import styles from './page.module.css'
-
-const inter = Inter({ subsets: ['latin'] })
 
 export default function Home() {
 return   <main>
-    <h1 className='text-3xl font-bold' >Hello World</h1>
+    <h1 className='text-3xl font-bold text-red-500' >Hello World</h1>
```
Comment (rkdixit96) [inline src/app/page.jsx:4]: Move text to screen text

File: src/components/AadharForm.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  const router = useRouter();
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    // advanced aadhar validation regex with optional space
+    const validationRegex = /^\d{4}\s?\d{4}\s?\d{4}$/;
+
+    if (!validationRegex.test(input)) {
+      alert('Please enter a valid Aadhar Number');
+      router.reload(window.location.pathname)
```
Comment (rkdixit96) [inline src/components/AadharForm.js:22]: Lets remove router - Just clear state if error

File: src/components/AadharForm.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+  const [input, setInput] = useState({ aadharInput: '' });
+  const [valid, setValid] = useState(false);
+
+  const router = useRouter();
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    // advanced aadhar validation regex with optional space
+    const validationRegex = /^\d{4}\s?\d{4}\s?\d{4}$/;
```
Comment (rkdixit96) [inline src/components/AadharForm.js:18]: Move regex to config file

File: src/components/AadharForm.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    if (!validationRegex.test(input)) {
+      alert('Please enter a valid Aadhar Number');
+      router.reload(window.location.pathname)
+    }
+    else {
+      // window.location.href='/onboarding/user/otp'
+      setValid(true);
+    }
+
+    console.log(input);
```
Comment (rkdixit96) [inline src/components/AadharForm.js:29]: Remove this

File: src/components/AadharForm.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+                  <input class="w-full border rounded p-2 outline-none focus:shadow-outline" type="OTPInput" name="OTPInput" id="OTPInput" placeholder="Enter OTP"></input>
+                </div>
+                <div class="mb-6 ">
+                  <label for="OTP" class="w-full block text-xs mb-1">Enter Mobile Number</label>
+                  <input class="w-full border rounded p-2 outline-none focus:shadow-outline" type="mobileInput" name="mobileInput" id="mobileInput" placeholder="Enter Mobile Number"></input>
+                </div>
+                <div class='flex justify-evenly w-full'>
+                  <button class="bg-black hover:bg-green-700 text-white  text-lg font-semibold px-4 py-2 rounded">Back</button>
+                  <button class="bg-green-700 hover:bg-black text-white  text-lg font-semibold px-4 py-2 rounded">Submit</button>
+                </div>
```
Comment (rkdixit96) [inline src/components/AadharForm.js:60]: Split into components - This is too big

File: src/components/Footer.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  
+        <div class="mb-6">
+          <h5 class="font-bold font-sans text-2xl mb-2.5">Patient Locker</h5>
+  
+          <ul class="list-none mb-0">
+            <li>
+              {/* <a href="#!" class="text-black text-lg">Link 1</a> */}
+              <Image src={logo} alt="ABDM QR" class='w-16 mx-auto items-center'/>
+            </li>
+
```
Comment (rkdixit96) [inline src/components/Footer.js:54]: Split into components - File is too big

**Review by rkdixit96 — CHANGES_REQUESTED** [review]

**Comment by ishit-007** [comment]
Patient Onboarding PR

**Pull request merged**